api/app/Policies/UserPolicy.php

```diff
@@ -80,8 +92,8 @@ public function update(User $user, User $model)
      */
     public function delete(User $user, User $model)
     {
-        if($user->isAdmin()){
-            if($user->id !== $model->id){
+        if ($user->isAdmin()) {
```

Comment:
I believe this needs to be changed to $user->hasRole(`platform_admin`)

Reply:
Agreed, however this is being worked on in https://github.com/GCTC-NTGC/gc-digital-talent/pull/5858

The only reason this touches the `UserPolicy` is because the `assign-any-role` happens on a user rather than a role so it made more sense here.

Reply:
ok got you 👍